frontend/internal-packages/agent/src/getMessages.ts

```diff
@@ -0,0 +1,36 @@
+import { type BaseMessage, isBaseMessage } from '@langchain/core/messages'
+import type { WorkflowConfigurable } from './chat/workflow/types'
+
+export async function getMessages(config: {
+  configurable: WorkflowConfigurable
+}): Promise<BaseMessage[]> {
+  const { thread_id, repositories } = config.configurable
+  const { checkpointer } = repositories.schema
+
+  if (!thread_id) {
+    return []
+  }
+
+  const checkpointConfig = {
+    configurable: {
+      thread_id,
+    },
+  }
+
+  const checkpointTuple = await checkpointer.getTuple(checkpointConfig)
+  if (!checkpointTuple || !checkpointTuple.checkpoint) {
+    return []
+  }
+
+  const channelValues = checkpointTuple.checkpoint.channel_values
+  if (!channelValues || !channelValues['messages']) {
+    return []
+  }
+
+  const messagesData = channelValues['messages']
+  if (!Array.isArray(messagesData)) {
+    return []
+  }
+
+  return messagesData.filter(isBaseMessage)
+}
```

Comment:
Of course we can leave it as it is this time since we can think about it later, but what I was thinking about after the migration from timeline_items to messages is to restore the state from checkpointer and use it as a UI. I am thinking that perhaps schema and artifact do not need to be prepared as their own database tables.